### PR TITLE
new option for raw audio parameters

### PR DESCRIPTION
**Pull request by eblanca**
I created this since sometimes I need sndfile-resample for resampling raw audio files, and those were not supported.
I added the command line option `-r' for providing all the needed parameters.
Usage guide is updated as well.
Just, remains that check for command line parameter
https://github.com/eblanca/sndfile-tools/blob/72e236855a5292b3a964c285ce8ea0193c1abbc6/src/resample.c#L55 which I updated but may be better written, yet I don't know how you figure out it.

### CODE REVIEW

File: src/resample.c [hunk 1]
```diff
@@ -51,7 +52,7 @@ main (int argc, char *argv [])
 		exit (0) ;
 		} ;
 
-	if (argc != 5 && argc != 7 && argc != 8)
+	if (argc != 5 && argc != 7 && argc != 8 && argc != 9 && argc != 10 && argc != 11)
```
Comment (erikd) [inline src/resample.c]: Maybe we need `argc < 5 || argc == 6 || argc > 11`.

File: src/resample.c [hunk 2]
```diff
@@ -35,7 +35,8 @@ main (int argc, char *argv [])
 	int normalize = 1 ;
 	sf_count_t	count ;
 	double		src_ratio = -1.0, gain = 1.0 ;
-	int			new_sample_rate = -1, k, converter, max_speed = SF_FALSE ;
+	int			new_sample_rate = -1, k, converter, max_speed = SF_FALSE, raw_bits ;
```
Comment (erikd) [inline src/resample.c]: Now is a good time to fix the indentation on that line.

Indentation should be with tabs where a single tab is the same as four spaces.

File: src/resample.c [hunk 3]
```diff
@@ -74,6 +75,48 @@ main (int argc, char *argv [])
...
 			converter = parse_int_or_die (argv [k], "converter") ;
 			}
+		else if (strcmp (argv [k], "-r") == 0)
+		{   k ++ ;
+            if (sscanf (argv [k], "%d,%d,%c,%d", &sfinfo.samplerate, &sfinfo.channels, &raw_type, &raw_bits) != 4)
+            {   printf ("Error : unable to decode raw audio parameters.\n") ;
+                usage_exit (argv [0]) ;
+                }
+            sfinfo.format = SF_FORMAT_RAW;
+            if (raw_type == 'i' && ( raw_bits == 8 || raw_bits == 16 || raw_bits == 24 || raw_bits == 32 ) )
```
Comment (erikd) [inline src/resample.c]: I am seeing some new code for handling command line args, but no update to the help output. I think a help output update is needed.

File: src/resample.c [hunk 4]
```diff
@@ -74,6 +75,48 @@ main (int argc, char *argv [])
...
+                {   case 32:
+                        sfinfo.format |= SF_FORMAT_FLOAT;
+                        break;
+
+                    case 64:
+                        sfinfo.format |= SF_FORMAT_DOUBLE;
+                        break;
+                    }
+                }
+		    else
```
Comment (erikd) [inline src/resample.c]: Weird indentation in that `else`.

Comment (eblanca) [reply src/resample.c]: it's due to mixture of tabs/spaces, I bet

File: src/resample.c (hunk 3 above)
Comment (eblanca) [reply src/resample.c]: what about https://github.com/eblanca/sndfile-tools/blob/1375f19ee95e1f7d362226ff651c5a4fe6681307/src/resample.c#L344 ?

File: src/resample.c (hunk 2 above)
Comment (eblanca) [reply src/resample.c]: fixed

File: src/resample.c (hunk 1 above)
Comment (eblanca) [reply src/resample.c]: done

**Comment by erikd** [comment]
Closing in favor of https://github.com/erikd/sndfile-tools/pull/63 (just some minor tweaks).

**Pull request closed without merging**